torch/utils/data/dataloader.py

```diff
@@ -319,17 +319,11 @@ def _shutdown_workers(self):
             if not self.shutdown:
                 self.shutdown = True
                 self.done_event.set()
-                # if worker_manager_thread is waiting to put, make place for it
-                try:
-                    while not self.data_queue.empty():
-                        self.data_queue.get()
-                except FileNotFoundError:
-                    # FileNotFoundError can happen when we rebuild the fd
-                    # fetched from the queue but the socket is already closed
-                    # from the worker side (e.g. due to Python shutting down).
-                    pass
                 for q in self.index_queues:
                     q.put(None)
+                # worker_manager_thread puts data into a queue.Queue which has
+                # infinite capacity. So we don't need to call get to make space
```

Comment:
Where did you find that info, I was pretty sure it uses a pipe underneath, so it has a constant limit. Also, why does it cause the import failure?

Reply:
It reads data from a `mp.SimpleQueue`, which workers put data into, but put processed data (pinned) in a `queue.Queue` (`self.data_queue` ). See line 221 above.

When there is no `worker_manager_thread`, we are `get`ting from a `SimpleQueue`, and the unpickler somehow cause importlib and causes the error when loading (https://github.com/python/cpython/blob/3.6/Lib/multiprocessing/queues.py#L337).

Reply:
here is the doc for `queue.Queue` that talks about infinite capacity: https://docs.python.org/2/library/queue.html#Queue.Queue

Reply:
Although Cpython indeed has a max size `SEM_VALUE_MAX`, but I think we can pretend that doesn't exist https://github.com/python/cpython/blob/3.6/Lib/multiprocessing/queues.py#L37-L39.

or do you think that I should still do the `get` when `worker_manager_thread` exists (and `data_queue` is not a `SimpleQueue`)?

Reply:
Ok I see what you meant now. Still, some of the workers might be blocked on a full queue in our direction, so we should still try to empty it, or they will deadlock, right? Couldn't we just expand the `except` case to cover all exceptions and add a comment that it's like this because of some obscure Python bugs?